repositories/productRepository.go

```diff
@@ -13,67 +16,173 @@ type ProductRepository struct {
 	DB *pgxpool.Pool
 }
 
-func (repository *ProductRepository) GetAllProducts() ([]models.Product, error) {
-	var products []models.Product = []models.Product{}
-	rows, err := repository.DB.Query(context.Background(), "SELECT * FROM products")
+func (repository *ProductRepository) GetAllProducts(employeeIdc string) ([]models.Product, error) {
+	earConstraints := []models.EarConstraint{}
+
+	query := `select ear.id "idear", ear.idrc, ear.idsc, p.name "property", o2.name "operator", ac.property_value from external_access_rights ear
+	left outer join access_constraints ac on ear.id = ac.idear
+	left outer join operators o2 on o2.id = ac.operator_id 
+	left outer join properties p on p.id = ac.property_id 
+	where ear.idrc = $1 and ear.r = true and ear.approved = true;`
+
+	rows, err := repository.DB.Query(context.Background(), query, employeeIdc)
 	defer rows.Close()
 	if err != nil {
 		return nil, err
 	}
 	for rows.Next() {
+		var earConstraint models.EarConstraint
+		err := rows.Scan(&earConstraint.Idear, &earConstraint.Idrc, &earConstraint.Idsc, &earConstraint.Property, &earConstraint.Operator, &earConstraint.PropertyValue)
+		if err != nil {
+			return nil, err
+		}
+		earConstraints = append(earConstraints, earConstraint)
+	}
+
+	var finalQuery strings.Builder
+
+	finalQuery.WriteString("select * from products p where p.idc = $1")
+
+	for _, earc := range earConstraints {
+		finalQuery.WriteString(" union select * from products p where p.idc = '" + earc.Idsc + "' ")
```

Comment:
After a lot of thought, I guess this approach is ok for a project of this size. But I have some concerns if this would be performant on a larger scale. Instead of unions and multiple selects, you might do something like: select * from product where (conditions1) and (conditions2) and ... (conditionsN). As you can see, there is only one select with a lot of eliminating conditions in the brackets.